From: Ximin Luo <infinity0@debian.org>
Date: Thu, 5 Sep 2024 15:14:01 +0200
Subject: Prevent spurious rebuilds

Forwarded: not-needed
---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index 00dd308..8699021 100644
--- a/Makefile
+++ b/Makefile
@@ -744,6 +744,7 @@ $(STARTUP_FILES): $(INCLUDE_DIRS) $(LIBC_BOTTOM_HALF_CRT_OBJS)
 	#
 	mkdir -p "$(SYSROOT_LIB)"
 	cp $(LIBC_BOTTOM_HALF_CRT_OBJS) "$(SYSROOT_LIB)"
+	touch $@
 
 	# Stamp the startup file installation.
 	@mkdir -p $(@D)
@@ -768,6 +769,7 @@ endif
 endif
 
 libc_so: $(INCLUDE_DIRS) $(LIBC_SO)
+	touch $@
 
 STATIC_LIBS = \
     $(SYSROOT_LIB)/libc.a \
@@ -788,6 +790,7 @@ STATIC_LIBS += \
 endif
 
 libc: $(INCLUDE_DIRS) $(STATIC_LIBS)
+	touch $@
 
 DUMMY := m rt pthread crypt util xnet resolv
 DUMMY_LIBS := $(patsubst %,$(SYSROOT_LIB)/lib%.a,$(DUMMY))
@@ -804,6 +807,7 @@ finish: $(STARTUP_FILES) libc $(DUMMY_LIBS)
 	#
 	# The build succeeded! The generated sysroot is in $(SYSROOT).
 	#
+	touch $@
 
 ifeq ($(LTO),no)
 # The check for defined and undefined symbols expects there to be a heap
@@ -1021,5 +1025,6 @@ clean:
 	$(RM) -r "$(BINDING_WORK_DIR)"
 	$(RM) -r "$(OBJDIR)"
 	$(RM) -r "$(SYSROOT)"
+	$(RM) -f include_dirs startup_files libc finish
 
 .PHONY: default libc libc_so finish install clean check-symbols bindings
