From: Ximin Luo <infinity0@debian.org>
Date: Thu, 5 Sep 2024 15:14:01 +0200
Subject: Prevent spurious rebuilds

Forwarded: not-needed
---
 Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Makefile b/Makefile
index 0d2a314..f8d4ed2 100644
--- a/Makefile
+++ b/Makefile
@@ -738,12 +738,15 @@ ifeq ($(WASI_SNAPSHOT), p2)
 		> "$(SYSROOT_INC)/__wasi_snapshot.h"
 endif
 
+	touch $@
+
 startup_files: include_dirs $(LIBC_BOTTOM_HALF_CRT_OBJS)
 	#
 	# Install the startup files (crt1.o etc).
 	#
 	mkdir -p "$(SYSROOT_LIB)" && \
 	cp $(LIBC_BOTTOM_HALF_CRT_OBJS) "$(SYSROOT_LIB)"
+	touch $@
 
 # TODO: As of this writing, wasi_thread_start.s uses non-position-independent
 # code, and I'm not sure how to make it position-independent.  Once we've done
@@ -763,6 +766,7 @@ endif
 endif
 
 libc_so: include_dirs $(LIBC_SO)
+	touch $@
 
 STATIC_LIBS = \
     $(SYSROOT_LIB)/libc.a \
@@ -779,6 +783,7 @@ STATIC_LIBS += \
 endif
 
 libc: include_dirs $(STATIC_LIBS)
+	touch $@
 
 dummy_libs:
 	#
@@ -793,6 +798,7 @@ finish: startup_files libc dummy_libs
 	#
 	# The build succeeded! The generated sysroot is in $(SYSROOT).
 	#
+	touch $@
 
 # The check for defined and undefined symbols expects there to be a heap
 # alloctor (providing malloc, calloc, free, etc). Skip this step if the build
@@ -987,5 +993,6 @@ clean:
 	$(RM) -r "$(BINDING_WORK_DIR)"
 	$(RM) -r "$(OBJDIR)"
 	$(RM) -r "$(SYSROOT)"
+	$(RM) -f include_dirs startup_files libc finish
 
 .PHONY: default startup_files libc libc_so dummy_libs finish install include_dirs clean check-symbols bindings
